compactor_scheduler/src/local_scheduler.rs

```diff
@@ -156,11 +327,315 @@ mod tests {
+                create: vec![created_1.into()],
+            };
+
+            let _ = scheduler
+                .job_status(CompactionJobStatus {
+                    job,
+                    status: CompactionJobStatusVariant::Update(commit_update),
+                })
+                .await;
+        }
+    }
+
+    #[tokio::test]
+    async fn test_status_update_upgrade() {
+        test_helpers::maybe_start_logging();
+
+        let (scheduler, existing_1, existing_2) = create_scheduler_with_partitions().await;
+        let jobs = scheduler.get_jobs().await;
+        let job = jobs
+            .into_iter()
+            .find(|job| job.partition_id == existing_1.partition.partition.id)
+            .unwrap();
+
+        let commit_update = CommitUpdate {
+            partition_id: job.partition_id,
+            delete: vec![],
+            upgrade: vec![existing_1.into(), existing_2.into()],
+            target_level: data_types::CompactionLevel::Final,
+            create: vec![],
+        };
+
+        assert!(scheduler
+            .job_status(CompactionJobStatus {
+                job,
+                status: CompactionJobStatusVariant::Update(commit_update),
+            })
+            .await
+            .is_ok());
+    }
+
+    #[tokio::test]
+    async fn test_status_update_can_replace_and_upgrade_at_once() {
+        test_helpers::maybe_start_logging();
+
+        let (scheduler, existing_1, existing_2) = create_scheduler_with_partitions().await;
+        let jobs = scheduler.get_jobs().await;
+        let job = jobs
+            .into_iter()
+            .find(|job| job.partition_id == existing_1.partition.partition.id)
+            .unwrap();
+
+        let created = ParquetFileBuilder::new(1002)
+            .with_partition(job.partition_id.get())
+            .build();
+
+        let commit_update = CommitUpdate {
+            partition_id: job.partition_id,
+            delete: vec![existing_1.into()],
+            upgrade: vec![existing_2.into()],
+            target_level: data_types::CompactionLevel::Final,
+            create: vec![created.into()],
+        };
+
+        assert!(scheduler
+            .job_status(CompactionJobStatus {
+                job,
+                status: CompactionJobStatusVariant::Update(commit_update),
+            })
+            .await
+            .is_ok());
+    }
+
+    #[tokio::test]
+    async fn test_status_skip() {
+        test_helpers::maybe_start_logging();
+
+        let (scheduler, _, _) = create_scheduler_with_partitions().await;
+        let jobs = scheduler.get_jobs().await;
+
+        for job in jobs {
+            assert!(scheduler
+                .job_status(CompactionJobStatus {
+                    job,
+                    status: CompactionJobStatusVariant::RequestToSkip(SkipReason::CompactionError(
+                        "some error".into()
+                    )),
+                })
+                .await
+                .is_ok());
```

Comment:
And likewise, in tests that send something other than `CompactionJobStatusVariant::Update`, I think it would be worthwhile to ensure we're getting an `Ack`:

```suggestion
            assert_matches!(scheduler
                .job_status(CompactionJobStatus {
                    job,
                    status: CompactionJobStatusVariant::RequestToSkip(SkipReason::CompactionError(
                        "some error".into()
                    )),
                })
                .await,
                Ok(CompactionJobStatusResult::Ack),
            );
```

Reply:
Done